File: RepositorioAcademico.Server/Domain/Rol.cs
namespace RepositorioAcademico.Server.Domain
{
    public class Rol
    {
        public int Id { get; set; }

        public string Nombre { get; set; } = string.Empty;

        public string? Descripcion { get; set; }

        public string Estado { get; set; } = "Activo";

        public ICollection<UsuarioRol> UsuarioRoles { get; set; } = [];

        public ICollection<RolPermiso> RolPermisos { get; set; } = [];
    }
}

File: repositorioacademico.client/src/app/models/rol.ts
export interface RolResumen {
  id: number;
  nombre: string;
  descripcion?: string | null;
  estado: string;
}

export interface Rol {
  id: number;
  nombre: string;
  descripcion?: string | null;
  estado: string;
  permisos: Permiso[];
}

export interface Permiso {
  id: number;
  codigo: string;
  descripcion: string;
  estado: string;
}
